data/wikipron/src/scrape.py

```diff
@@ -59,9 +59,20 @@ def _build_scraping_config(
     _call_scrape(config_settings, phonetic_config, phonetic_path)
 
 
-def main() -> None:
+def main(args) -> None:
     with open(LANGUAGES_PATH, "r") as source:
         languages = json.load(source)
+
+    ###Checks restriction codes are valid.
+    if args.restriction: 
+        for key in args.restriction:
+            try:
+                assert key in languages 
+            except AssertionError:
+                print("Language restrictions must be valid ISO code.")
```

Comment:
Could you change this to `logging.info()` and log the failed `key`?

Reply:
This should also be a fatal error. While `exit()` does kill things it just makes it look like it succeeded because the return code of the program will be 0 (non-zero means "failure"). I would use `logging.fatal` instead and then `exit(1)` or something like that.